aesm-client/src/windows/win_aesm_client.rs

```diff
@@ -0,0 +1,393 @@
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_whitelist_size: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            white_list_size: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_white_list: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            white_list: *mut u8,
+            buf_size: u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_sec_domain_id: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            sec_domain_id: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    switch_sec_domain: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            sec_domain_id: u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_epid_provision_status: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            epid_pr_status: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+    get_platform_service_status: Option<
+        unsafe extern "system" fn(
+            this: *mut AesmInterface,
+            pse_status: *mut u32,
+            result: *mut AesmError,
+        ) -> HRESULT,
+    >,
+}
+
+const CLSID_AESMINTERFACE: CLSID = CLSID {
+    Data1: 0x82367CAB,
+    Data2: 0xF2B9,
+    Data3: 0x461A,
+    Data4: [0xB6, 0xC6, 0x88, 0x9D, 0x13, 0xEF, 0xC6, 0xCA],
+};
+const IID_IAESMINTERFACE: IID = IID {
+    Data1: 0x50AFD900,
+    Data2: 0xF309,
+    Data3: 0x4557,
+    Data4: [0x8F, 0xCB, 0x10, 0xCF, 0xAB, 0x80, 0x2C, 0xDD],
+};
+
+trait HresultExt:Sized {
+    fn into_io_error(self) -> IoResult<Self>;
+}
+
+impl HresultExt for HRESULT {
+    fn into_io_error(self) -> IoResult<HRESULT> {
+        if SUCCEEDED(self) {
+            Ok(self)
+        } else {
+            Err(IoError::from_raw_os_error(self))
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct AesmClient {
+    interface: *mut AesmInterface,
+}
+
+impl Drop for AesmClient {
+    fn drop(&mut self) {
+        unsafe {
+            if let Some(release) = (*(*self.interface).vtbl).release {
+                release(self.interface);
+            }
+            CoUninitialize();
+        }
+    }
+}
+
+impl AesmClient {
+    pub fn new() -> Result<Self> {
+        AesmClient::create_instance()
+    }
+
+    fn initialize_com(&self) -> Result<HRESULT> {
+        unsafe {
+            CoInitializeEx(
+                ptr::null_mut(),
+                COINIT_MULTITHREADED | COINIT_DISABLE_OLE1DDE,
+            )
+                .into_io_error()
+                .map_err(|e| Error::AesmCommunication(IoError::new(ErrorKind::Other, format!("Failed to initialize COM: {}", e))) )
+        }
+    }
+
+    pub fn try_connect(&self) -> Result<()> {
+        self.initialize_com().map(|_| ())
```

Comment:
This doesn't make sense. You will already have called `create_instance` when you get here, which already did this. But see our Slack discussion for possible change in architecture.